retrieve/retrieve.go

```diff
@@ -0,0 +1,99 @@
+package retrieve
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/joshuatcasey/collections"
+	"github.com/paketo-buildpacks/libdependency"
+	"github.com/paketo-buildpacks/libdependency/versionology"
+	"github.com/paketo-buildpacks/libdependency/workflows"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"golang.org/x/exp/slices"
+)
+
+type GenerateMetadataFunc func(version versionology.HasVersion) (cargo.ConfigMetadataDependency, error)
+
+func NewMetadata(id string, getNewVersions libdependency.HasVersionsFunc, generateMetadata GenerateMetadataFunc, targets ...string) {
```

Comment:
Again some comments here would be really useful!

Reply:
comments added